### PR TITLE
Inital commit of azure function

### PR DESCRIPTION
**Pull request by MathewBeldon**
Initial PR

### CODE REVIEW

File: Best Practices/Challenges/AzureFunction/AzureFunction.Challenge.Persistence/DatabaseInitialiser.cs [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+        private readonly AzureFunctionDbContext _context;
+
+        public DatabaseInitialiser(AzureFunctionDbContext context)
+        {
+            _context = context;
+        }
+
+        public void Initialize()
+        {
+            _context.Database.EnsureCreated();
```
Comment (MathewBeldon) [inline Best Practices/Challenges/AzureFunction/AzureFunction.Challenge.Persistence/DatabaseInitialiser.cs:14]: https://github.com/stephen-atkinson/recipes-api-base-build/blob/main/Recipes.Core/Application/Extensions/HostExtensions.cs

File: Best Practices/Challenges/AzureFunction/AzureFunction.Challenge.Core/Services/ProductService.cs [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    public class ProductService : IProductService
+    {
+        private readonly IProductRepository _productRepository;
+
+        public ProductService(IProductRepository productRepository)
+        {
+            _productRepository = productRepository;
+        }
+
+        public async Task<IEnumerable<ProductDto>> GetProductsAsync()
```
Comment (MathewBeldon) [inline Best Practices/Challenges/AzureFunction/AzureFunction.Challenge.Core/Services/ProductService.cs:15]: IReadOnlyCollection call ToArray on Select

File: Best Practices/Challenges/AzureFunction/AzureFunction.Challenge.Core/Interfaces/IOrderRepository.cs [hunk 3]
```diff
@@ -0,0 +1,11 @@
+﻿using AzureFunction.Challenge.Core.Models;
```
Comment (MathewBeldon) [inline Best Practices/Challenges/AzureFunction/AzureFunction.Challenge.Core/Interfaces/IOrderRepository.cs:1]: RM repository

File: Best Practices/Challenges/AzureFunction/README.MD [hunk 4]
```diff
@@ -0,0 +1,202 @@
+# BurgetBytes Azure Function
+
```
Comment (MathewBeldon) [inline Best Practices/Challenges/AzureFunction/README.MD:2]: Add how to run the function

File: Best Practices/Challenges/AzureFunction/AzureFunction.Challenge.Core/Models/Order.cs [hunk 5]
```diff
@@ -0,0 +1,12 @@
+﻿namespace AzureFunction.Challenge.Core.Models
+{
+    public class Order
+    {
+        public int Id { get; set; }
+        public string DeliveryAddress { get; set; }
```
Comment (MathewBeldon) [inline Best Practices/Challenges/AzureFunction/AzureFunction.Challenge.Core/Models/Order.cs:6]: Create address object

Comment (MathewBeldon) [reply Best Practices/Challenges/AzureFunction/AzureFunction.Challenge.Core/Models/Order.cs:6]: Owned types

**Comment by stephen-atkinson** [comment]
issue (blocking): Need to move this out of the Best Practices folder

File: Azure Functions/Challenges/AzureFunction.Challenge/Core/Services/OrderService.cs [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+            };
+
+
+            var entityEntry = await _context.Orders.AddAsync(order);
+            await _context.SaveChangesAsync();
+
+            return new OrderResponseDto
+            {
+                Id = entityEntry.Entity.Id,
+                DeliveryTime = GetDeliveryTime()
```
Comment (stephen-atkinson) [inline Azure Functions/Challenges/AzureFunction.Challenge/Core/Services/OrderService.cs:63]: Need to persist DeliveryTime
Add new migration for single column
Save to DB
Return as part of the GetOrder endpoint

**Review by stephen-atkinson — CHANGES_REQUESTED** [review]

**Review by stephen-atkinson — APPROVED** [review]

**Pull request merged**